pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGeneratorFunctors.hpp

```diff
@@ -371,86 +355,249 @@ void applyGenMultiRZ(
         });
 }
 
+template <class PrecisionT, class FuncT> class applyNCGenerator1Functor {
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr;
+    const FuncT core_function;
+    KokkosIntVector indices;
+    KokkosIntVector parity;
+    KokkosIntVector rev_wires;
+    KokkosIntVector rev_wire_shifts;
+    std::size_t mask{0U};
+    static constexpr std::size_t one{1U};
+    std::size_t i0;
+    std::size_t i1;
+
+  public:
+    template <class ExecutionSpace>
+    applyNCGenerator1Functor([[maybe_unused]] ExecutionSpace exec,
+                             Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+                             std::size_t num_qubits,
+                             const std::vector<std::size_t> &controlled_wires,
+                             const std::vector<bool> &controlled_values,
+                             const std::vector<std::size_t> &wires,
+                             FuncT core_function_)
+        : arr(arr_), core_function(core_function_) {
+        const std::size_t n_contr = controlled_wires.size();
+        const std::size_t n_wires = wires.size();
+        const std::size_t nw_tot = n_contr + n_wires;
+        PL_ASSERT(n_wires == 1);
+        PL_ASSERT(num_qubits >= nw_tot);
+
+        std::vector<std::size_t> all_wires;
+        all_wires.reserve(nw_tot);
+        all_wires.insert(all_wires.begin(), wires.begin(), wires.end());
+        all_wires.insert(all_wires.begin(), controlled_wires.begin(),
+                         controlled_wires.end());
+
+        const auto &[parity_, rev_wires_] =
+            reverseWires(num_qubits, wires, controlled_wires);
+        parity = parity_;
+        std::vector<std::size_t> indices_ =
```

Comment:
any reason to add the underscore suffix to `inidces` here?

Reply:
Can we mark it as `const`?

Reply:
This `indices_` is used in a few places in `BasicGateFunctors.hpp` and `MatrixGateFunctors.hpp` (e.g. [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/bde61072f78f2fd59d77da52dbcc6affc383f25d/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp#L67)). As I mentioned in another comment, the convention used has been non-underscored objects are data members for the functors, and underscore is added for the objects (e.g. this std::vector here) to be copied into the functor. I can swap them around, but this will require changing all three files to be consistent?